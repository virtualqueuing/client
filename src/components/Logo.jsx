import { Link } from "react-router-dom";
import styled, { css } from "styled-components";
import logo from "../assets/Logo-v5.png";
import React from "react";

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const LogoImage = styled.img`
  margin: 3% 0;
  ${({ logoSize }) => {
    switch (logoSize) {
      case "small":
        return css`
          width: 120px;
          height: 60px;
        `;
      case "medium":
        return css`
          width: 150px;
          height: auto;
        `;
      case "large":
        return css`
          width: 180px;
          height: auto;
        `;
      default:
        return css``;
    }
  }}
`;

const Logo = ({ logoSize }) => {
  return (
    <Wrapper>
      <Link to="/home">
        <LogoImage logoSize={logoSize} src={logo} alt="logo for redirecting to home page" />
      </Link>
    </Wrapper>
  );
};

export default Logo;
