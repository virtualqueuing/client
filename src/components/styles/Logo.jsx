import styled from "styled-components";

export const Logo = styled.img`
  margin: 50px;
  width: 246px;
  height: 113px;
  @media (max-width: 1500px) {
    width: 150px;
    height: auto;
  }
`;
