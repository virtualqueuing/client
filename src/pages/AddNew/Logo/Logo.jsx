import styled from "styled-components";
import { fontSize, color } from "styled-system";

const Logotext = styled.h1`
  text-align: center;
  margin-bottom: 20px;

  ${fontSize};
  ${color};
`;

const Logo = () => {
  return (
    <Logotext color="white" fontSize="4x">
      Haidilao
    </Logotext>
  );
};

export default Logo;
