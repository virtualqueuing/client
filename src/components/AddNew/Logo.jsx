import { StyledLogo } from "../styles/AddNew.styles";

const Logo = () => <StyledLogo>Haidilao</StyledLogo>;

export default Logo;
