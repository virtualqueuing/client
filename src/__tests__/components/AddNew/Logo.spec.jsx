import Logo from '../../../components/Logo'
import React from 'react';
import { render, screen } from '@testing-library/react'
import logo from '../../../assets/Logo-v5.svg'

jest.mock("react-router-dom", () => ({
    Link: ({ to, children }) => (
        <div className="link" data-to={to}>{children}</div>
    )
}))

describe("<Logo>", () => {
    it("should present on the top of the queue list page", () => {
        const { container } = render(<Logo />) ;
        screen.debug(container)
    })
    it("should present large on the top of the queue list page", () => {
        const { container } = render(<Logo logoSize="large" />) ;
        screen.debug(container)
        // expect(container).toMatchSnapshot()
    })
    it("should present medium on the top of the queue list page", () => {
        const { container } = render(<Logo logoSize="medium" />) ;
        screen.debug(container)
        // expect(container).toMatchSnapshot()
    })
    it("should present small on the top of the queue list page", () => {
        const { container } = render(<Logo logoSize="small" />) ;
        screen.debug(container)
        // expect(container).toMatchSnapshot()
    })
})