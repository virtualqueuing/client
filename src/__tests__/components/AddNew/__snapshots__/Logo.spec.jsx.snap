// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo> should present large on the top of the queue list page 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin: 3% 0;
  width: 180px;
  height: auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="link"
    >
      <img
        class="c1"
        src="[object Object]"
      />
    </div>
  </div>
</div>
`;

exports[`<Logo> should present medium on the top of the queue list page 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin: 3% 0;
  width: 150px;
  height: auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="link"
    >
      <img
        class="c1"
        src="[object Object]"
      />
    </div>
  </div>
</div>
`;

exports[`<Logo> should present small on the top of the queue list page 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin: 3% 0;
  width: 120px;
  height: 60px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="link"
    >
      <img
        class="c1"
        src="[object Object]"
      />
    </div>
  </div>
</div>
`;
